README.md

```diff
@@ -1,41 +1,108 @@
+## Getting Started
 
-The following is the template for the final README.md file:
+Add the GeolocatorService nuget package to your project heads.
 
----
+To have access to reactive extensions such as ```GetAndObserveLocation```, add the GeolocatorService.Reactive nuget package to your project.
 
-# Project Title
+### Add the relevant permissions
 
-{Project tag line}
+#### UWP
 
-{Small description of the purpose of the project}
+Add the Location capability in your manifest.
 
-[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)
+#### iOS
 
-## Getting Started
+Add the NSLocationWhenInUsageDescription and NSLocationUsageDescription values to your info.plist file. This is the message that is displayed to your user when the permission for their location is requested. For instance,
+
+```
+<key>NSLocationWhenInUsageDescription</key>
+<string>TODO NSLocationWhenInUsageDescription</string>
+<key>NSLocationUsageDescription</key>
+<string>TODO NSLocationUsageDescription</string>
+```
+
+#### Android
+
+Add the location permission to your AssemblyInfo.cs file.
+
+```
+[assembly: UsesPermission(Android.Manifest.Permission.AccessFineLocation)]
+```
+
+### Request permission
+
+After you have added the relevant location permission info to your project configuration, locate the spot in your code where you want to ask the user for their location permission. This must be done before accessing their location. Add the following line to do so:
+
+```
+var isGranted = await locationService.RequestPermission(ct);
+```
+
+You may safely request the permission multiple times; the user will only get prompted as often as is necessary.
+
+For instance, if you ask the user for their permission once and they decline, calling ```RequestPermission``` again will not prompt the user again, but simply return ```false```.
+
+If the user answers "Allow once", then calling ```RequestPermission``` again during the lifetime of the app will simply return ```true```. Once the app is restarted, calling ```RequestPermission``` will prompt them for the permission once again, since their original permission has now expired.
 
-{Instructions to quickly get started using the project: pre-requisites, packages
-to install, sample code, etc.}
+Once permission is requested, you can use the various methods and extension methods of IGeolocatorService.
 
 ## Features
 
-{More details/listing of features of the project}
+### Request the user's location
+
+Once you have obtained the location permission, it's a simple matter to obtain the location:
+
+```
+var location = await locationService.GetLocation(ct);
+```
+
+If the user has denied the location permission, an ```InvalidOperationException``` is thrown; therefore, we recommend not calling this method if you know the user has denied the location permission and this is a normal scenario. If the app has not kept track of this information, we recommend calling this extension method instead:
```

Comment:
If the user has denied implies the user was requested the permission, but it's possible the user wasn't shown the prompt right?

We could say "if the user has not granted the location permission", would cover both scenarios.